### ISSUE TITLE
[4.3.0]: Entry source headings not being hidden when they should

### ISSUE THREAD

**Issue by AndrewMac** [craftcms/cms#12193]
### Description
My Craft installation has multiple sites with a number of sections that are available only to certain sites. In the "Entries" view, the entry sources are grouped under different headings. Before the just-released 4.3.0 version, if all the entry sources in a group were disabled (because they were not required for that site), the heading was hidden as well. In 4.3.0, though, the HTML has been changed (in a better way, actually!) but the heading now shows even if all the sources are correctly hidden. 

The result is that I now see a whole bunch of headings that used to be hidden.

Basically, it looks like the heading is not picking up the "hidden" class, as you will see in the code I will post below.

### Old behaviour
```
<li class="heading hidden>…</li>
<li class="hidden">…</li>
<li class="hidden">…</li>
```

### New improved behaviour
```
<li class="heading"> <!-- should also have picked up the "hidden" class" -->
    <span>…</span>
    <ul>
        <li class="hidden">…</li>
        <li class="hidden">…</li>
    </ul>
</li>
```

### Craft CMS version

4.3.0 Pro

### Installed plugins and versions

-

**Comment by brandonkelly** [comment]
Thanks for reporting that! Fixed for the next release.

**Comment by brandonkelly** [comment]
Craft 4.3.1 is out with the fix for this.
